workspaces/adventure-pack/goodies/python3/src/transpose_matrix/test.py

```diff
@@ -0,0 +1,45 @@
+from . import *
+import pytest
+
+
+def test_improper_input() -> None:
+    with pytest.raises(ValueError):
+        transpose_matrix([])
+
+    with pytest.raises(ValueError):
+        transpose_matrix([[], []])
+
+    with pytest.raises(ValueError):
+        transpose_matrix([[0], [0, 1]])
+
+
+def test_nonsymetric_transpose() -> None:
+    input_1 = [[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11]]
+    expected_1 = [[0, 4, 8], [1, 5, 9], [2, 6, 10], [3, 7, 11]]
+    input_2 = [
```

Comment:
Let's go a step further and group the asserts too, so the entirety of that test case is co-located!

```suggestion
    input_1 = [[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11]]
    expected_1 = [[0, 4, 8], [1, 5, 9], [2, 6, 10], [3, 7, 11]]
    assert transpose_matrix(input_1) == expected_1
    
    input_2 = [
```